docs/tutorials/gallery_adding_an_assistant.py

```diff
@@ -0,0 +1,110 @@
+"""
+# Adding an Assistant
+
+Ragna has builtin support for [several assistants][ragna.assistants], but there may be cases where you 
+want to use one that is not currently supported.
+
+This tutorial walks you through the basics of creating an assistant for LLMs that are not
+currently supported. 
+"""
+
+# %%
+# ## The Finished Product
+
+# %%
+# !!! note
+#
+#     The code snippet below only includes up to Step 1. To actually include your source
+#     storage in Ragna, you must still perform Step 2.
+
+from ragna.core import Assistant
+
+
+class TutorialAssistant(Assistant):
+    """
+    This is a basic assistant created for didactic purposes
+    """
+
+    def answer(self, prompt, sources):
+        """Answer a prompt given some sources.
+
+        Args:
+            prompt: Prompt to be answered.
+            sources: Sources to use when answering answer the prompt.
+
+        Returns:
+            Answer.
+        """
+        yield (
+            f"This is a default answer. There were {len(sources)} sources."
+            ""
+            f"The prompt was"
+            f"{prompt}"
+        )
+
+
+# %%
+# ## The Explanation
+
+# %%
+# !!! tip
+#
+#     For organizational purposes, this tutorial is divided into steps, but you may wish to perform
+#     Step 2 during Step 1 for manual testing or debugging.
+
+
+# %%
+# ### Step 0: Import Necessary Modules
+
+# %%
+# First start by importing some necessary components.
```

Comment:
You misunderstood my comment in https://github.com/Quansight/ragna/pull/358#discussion_r1526158304. Type hints are really useful for tutorials. Please put them back. What I meant is that you don't need to explain the `typing` import.